Build error using CUDA backend

# Summary
I'm trying to build the RNG using the CUDA backend, and am getting the following error:
```
/opt/modulefiles/dpcpp/dpcpp_wrapper/clang++ -fPIC -O3 -DNDEBUG -fsycl -fsycl-targets=nvptx64-nvidia-cuda -shared -Wl,-soname,libonemkl.so.0 -o ../lib/libonemkl.so.0 rng/CMakeFiles/onemkl_rng.dir/rng_loader.cpp.o  -lSYCL_LIBRARY-NOTFOUND 
/usr/bin/ld: cannot find -lSYCL_LIBRARY-NOTFOUND
```

# Version
I'm using tip of the develop branch (da62115), and llvm w/ cuda from git hash 8b9d2106af0d of the sycl branch of llvm.
CUDA is version 11.8.0, and gcc is 11.2.0
cmake is 3.23.2

the OS is centos 7.

# Environment
cmake command is:
```
>  cmake ../src/ -DCMAKE_CXX_COMPILER=clang++ -DENABLE_CUBLAS_BACKEND=False  -DENABLE_MKLCPU_BACKEND=False -DENABLE_MKLGPU_BACKEND=False -DENABLE_CURAND_BACKEND=True --install-prefix=/opt/oneMKL/2023.05.13-da62115-cuda -DTARGET_DOMAINS=rng

-- CMAKE_BUILD_TYPE: None, set to Release by default
-- The CXX compiler identification is Clang 17.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/modulefiles/dpcpp/dpcpp_wrapper/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- TARGET_DOMAINS: rng
-- Looking for dpc++
-- Performing Test is_dpcpp
-- Performing Test is_dpcpp - Success
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found CUDA: /opt/cuda/11.8.0 (found suitable version "11.8", minimum required is "10.0") 
-- Found cuRAND: /opt/cuda/11.8.0/include  
-- The C compiler identification is Clang 17.0.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/modulefiles/dpcpp/dpcpp_wrapper/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
CMake Deprecation Warning at deps/googletest/CMakeLists.txt:53 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Found PythonInterp: /usr/bin/python (found version "2.7.5") 
-- SYCL_DEVICE_FILTER will be set to the following value(s): [cuda:gpu] for run-time dispatching examples
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/oneMKL/bld/oneMKL/bld
```

the LD_LIBRARY_PATH is `/opt/dpcpp/2023.05.18-cuda-8b9d2106af0d/lib:/opt/cuda/11.8.0/extras/CUPTI/lib64:/opt/cuda/11.8.0/lib64:/opt/gcc/11.2.0/lib64:/opt/gcc/11.2.0/lib:/usr/lib64:/opt/cuda/11.8.0/lib64/stubs`

--- a/cmake/FindCompiler.cmake
+++ b/cmake/FindCompiler.cmake
@@ -27,7 +27,10 @@ check_cxx_compiler_flag("-fsycl" is_dpcpp)
 if(is_dpcpp)
   # Workaround for internal compiler error during linking if -fsycl is used
   get_filename_component(SYCL_BINARY_DIR ${CMAKE_CXX_COMPILER} DIRECTORY)
-  find_library(SYCL_LIBRARY NAMES sycl PATHS "${SYCL_BINARY_DIR}/../lib")
+  find_library(SYCL_LIBRARY NAMES sycl PATHS "${SYCL_BINARY_DIR}/../lib" "${SYCL_BINARY_DIR}/lib" ENV LIBRARY_PATH ENV PATH)
+  if(NOT SYCL_LIBRARY)
+    message(FATAL_ERROR "SYCL library is not found in ${SYCL_BINARY_DIR}/../lib, PATH, and LIBRARY_PATH")
+  endif()
 
   add_library(ONEMKL::SYCL::SYCL INTERFACE IMPORTED)
   if(UNIX)

